In [2]:
import numpy as np
import matplotlib.pyplot as plt
import yfinance as yf

In [3]:
tickers = ['AAPL', 'MSFT', 'NVDA', 'DIS']

prices = yf.download(tickers,start='2020-01-01', end='2025-08-27')['Close']

daily_rend = prices.pct_change().dropna()

C:\Users\samue\AppData\Local\Temp\ipykernel_12776\920268129.py:3: FutureWarning: YF.download() has changed argument auto_adjust default to True
  prices = yf.download(tickers,start='2020-01-01', end='2025-08-27')['Close']
[*********************100%***********************]  4 of 4 completed


In [4]:
#Rend, Riesgo Ind.
rend_mean = daily_rend.mean()*252
rend_mean

Ticker
AAPL    0.256734
DIS     0.019170
MSFT    0.256657
NVDA    0.752535
dtype: float64

In [5]:
rend_vol = daily_rend.std()*252**0.5
rend_vol

Ticker
AAPL    0.323874
DIS     0.338956
MSFT    0.301211
NVDA    0.540992
dtype: float64

In [10]:
w = np.array([0.25, 0.25, 0.25, 0.25])

cov = daily_rend.cov()

In [21]:
port_mean = float(np.dot(rend_mean, w))
port_mean

0.3212740302037045

In [22]:
port_vol = float(np.sqrt(w.T @ cov @ w) * np.sqrt(252))
port_vol

0.30763394107917036

In [23]:
daily_rend.corr()

Ticker      AAPL       DIS      MSFT      NVDA
Ticker                                        
AAPL    1.000000  0.456882  0.721735  0.589579
DIS     0.456882  1.000000  0.470806  0.396058
MSFT    0.721735  0.470806  1.000000  0.680905
NVDA    0.589579  0.396058  0.680905  1.000000

**Interpretación:**

- Todas las que son 1 es porque son la misma empresa, por lo que se mueven exactamente igual
- p altas: el 0.72 entre AAPL y MSFT refleja que ambas empresas están en el sector tecnológico y responden a factores comunes del mercado.
- Las tecnológicas (AAPL, MSFT, NVDA) tienen correlaciones más altas entre sí, por lo que se mueven en la misma dirección frecuentemente.
- Disney (DIS) tiene correlaciones más bajas con las demás, siendo la que aporta mayor diversificación al portafolio al estar en un sector diferente, porque no se mueve tan parecido a las tecnológicas.
- Este portafolio está concentrado en tecnología, lo que es riesgoso porque depende de un solo sector, y sus activos se mueven casi a la par. Solo DIS aporta un poco de diversificación, pero no lo suficiente. Aparte, todos pertenecen al S&P 500.